### PR TITLE
debounce updates

### PR DESCRIPTION
**Pull request by Ryan1729**
This attempts to address #41.

Note: I just pulled the constant `64` out of thin air. Maybe a command line argument to change it would make sense?

Also I'm not sure what the best way to test line 197 is. One option would be to remove the if statement, effectively resulting in a call to `clearTimeout(undefined)`, which doesn''t seem to cause any problems, but I couldn't find a definitive answer as to whether that's okay from [the docs](https://nodejs.org/api/timers.html#timers_cleartimeout_timeout).

### CODE REVIEW

File: source/elm-live.js [hunk 1]
```diff
@@ -183,10 +183,27 @@ ${indent(String(elmMake.error), 2)}
...
   };
 
+  // wait until the returned funtion has not been called for `wait`
+  // milliseconds before calling the passed in function
+  const debounce = (func, wait) => {
+    let timeout;
+    return (...theArgs) => {
+      const context = this;
+      const later = () => {
+        func.apply(context, theArgs);
```
Comment (tomekwi) [inline source/elm-live.js]: Any reason it’s not just `func(...theArgs)`? Since we use ES6 anyway, let’s just go for it! 😀

**Comment by Ryan1729** [comment]
Okay. Let me know if there's anything else I should do, (squashing commits, updating version numbers etc.)

File: test.js [hunk 2]
```diff
@@ -827,3 +828,28 @@ test('Prints any other `--before-build` command error', (assert) => new Promise(
...
+
+  const checkTimesRan = () => {
+    assert.is(timesRan, 1,
+      `only calls the passed function once when the returned function
+is called several times within the wait interval`
+    );
+    resolve();
+  };
+
+  setTimeout(checkTimesRan, wait + 1);
```
Comment (tomekwi) [inline test.js]: How about bumping the difference between the wait and the check up? I’m a bit wary of race conditions at such small intervals – and that would mean flaky tests. How about 200ms? It’s slow for a unit test, but we don’t _really_ have to worry about build duration. 😄

Comment (Ryan1729) [inline test.js]: I'd be surprised if incrementing an integer 4 times took even 5ms on anything resembling modern hardware, but better safe than sorry I suppose.

**Comment by tomekwi** [comment]
Thanks a lot for this man! Sorry for taking so long to merge this, a really crazy time in my life 😄

**Comment by tomekwi** [comment]
Published in 2.6.1.

**Pull request merged**